rust/azure_iot_operations_mqtt/src/session/wrapper.rs

```diff
@@ -74,11 +86,14 @@ impl Session {
 
         // Add AIO metric to user properties when using AIO MQTT broker features
         // TODO: consider user properties from being supported on SessionOptions or ConnectionSettings
-        let user_properties = if options.aio_broker_features {
-            vec![("metriccategory".into(), "aiosdk-rust".into())]
+        let mut user_properties = if options.aio_broker_features {
+            vec![("metriccategory".to_string(), "aiosdk-rust".to_string())]
```

Comment:
could you move
```
if options.persist {
            user_properties.push(("aio-persistence".to_string(), true.to_string()));
        }
```
into this if block since we won't set aio-persistence if aio_broker_features is false? Save the tiniest amount of processing in the other case :D